backend/src/backend/db/models.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime
+from ipaddress import IPv4Address
+from typing import List, Optional
+from uuid import UUID
+
+from sqlalchemy import DateTime, Enum, ForeignKey, UniqueConstraint, text
+from sqlalchemy.dialects.postgresql import ARRAY, CITEXT, INET
+from sqlalchemy.orm import (
+    DeclarativeBase,
+    Mapped,
+    MappedAsDataclass,
+    mapped_column,
+    relationship,
+)
+from sqlalchemy.sql.schema import MetaData
+
+from backend.enums import StatusEnum
+
+
+class Base(MappedAsDataclass, DeclarativeBase):
+    # This map details the specific transformation of types between Python and
+    # PostgreSQL. This is only needed for the case where a specific PostgreSQL
+    # type has to be used.
+
+    type_annotation_map = {  # noqa: RUF012
+        str: CITEXT,  # transform Python str to PostgreSQL CITEXT
+        List[str]: ARRAY(
+            item_type=CITEXT
+        ),  # transform Python List[str] into PostgreSQL Array of strings
+        datetime: DateTime(
+            timezone=False
+        ),  # transform Python datetime into PostgreSQL Datetime without timezone
+        IPv4Address: INET,  # transform Python IPV4Address into PostgreSQL INET
+    }
+
+    # This metadata specifies some naming conventions that will be used by
+    # alembic to generate constraints names (indexes, unique constraints, ...)
+    metadata = MetaData(
+        naming_convention={
+            "ix": "ix_%(column_0_label)s",
+            "uq": "uq_%(table_name)s_%(column_0_name)s",
+            "ck": "ck_%(table_name)s_%(constraint_name)s",
+            "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+            "pk": "pk_%(table_name)s",
+        }
+    )
+    pass
+
+
+class Country(Base):
+    __tablename__ = "country"
+
+    code: Mapped[str] = mapped_column(primary_key=True)
```

Comment:
We need to be explicit (comment) about what we'll store here.
I think we want [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) as that what seems to be commonly used (and we can rely on `pycountry` if we need conversion)